File: Week2/longestPalindrome/longestPalindrome.js
const longestPalindrome = s => {
    const count = {};
    let length = 0;
    let oddCount = 0;

    for (let i = 0; i < s.length; i++) {
        const currChar = s[i]
        if (count[currChar] !== undefined) {
            count[currChar] += 1;
        } else {
            count[currChar] = 1;
        }
    }

    if (Object.keys(count).length === 1) {
        return count[s[0]];
    }

    for (let i = 0; i < Object.values(count).length; i++) {
        const currVal = Object.values(count)[i];

        if (currVal > 1) {
            if (currVal % 2 === 0) {
                length += currVal
            } else {
                oddCount++;
                length += currVal - 1;
            }
        } else {
            oddCount++;
        }
        
    }

    if (oddCount > 0) {
        return length + 1; 
    }
    return length;
}

File: Week2/longestPalindrome/longestPalindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = {}
        length = 0
        oddCount = 0

        for char in s:
            if char in count:
                count[char] += 1
            else:
                count[char] = 1

        if len(count) == 1:
            return count[s[0]]
        
        for key in count:
            if count[key] > 1:
                if count[key] % 2 == 0:
                    length += count[key]
                else:
                    oddCount += 1
                    length += count[key] - 1
            else:
                oddCount += 1

        if oddCount > 0:
            return length + 1
        return length